Add Spectro-K8s-dashboard (#1106)
spectrocloud/librarium: content/docs/06-integrations/00-spectro-k8s-dashboard.md

---
title: 'Spectro Kubernetes Dashboard'

> metaTitle: 'Spectro Kubernetes Dashboard'

Reviewer: ```suggestion
metaTitle: 'Spectro Cloud's Integrated Kubernetes Dashboard'
```
For SEO purposes.

> metaDescription: 'Kubernetes Dashboard Monitoring pack in Spectro Cloud'

Reviewer: ```suggestion
metaDescription: 'The Integrated Kubernetes Dashboard Monitoring pack in Spectro Cloud reduces the complexity of standing up the Kubernetes dashboard for a cluster. '
```

hiddenFromNav: true
type: "integration"
hideToC: false

> category: ['Spectro Integrations']

Reviewer: ```suggestion
category: ['Spectro Integrations', 'monitoring']
```
We need both categories as in the UI today, users will find it through the monitoring category.

logoUrl: 'https://registry.spectrocloud.com/v1/k8s-dashboard/blobs/sha256:2de5d88b2573af42d4cc269dff75744c4174ce47cbbeed5445e51a2edd8b7429?type=image/png'
---

import Tabs from 'shared/components/ui/Tabs';
import WarningBox from 'shared/components/WarningBox';
import InfoBox from 'shared/components/InfoBox';
import PointsOfInterest from 'shared/components/common/PointOfInterest';
import Tooltip from "shared/components/ui/Tooltip";

# Spectro Kubernetes Dashboard

> Spectro Kubernetes Dashboard is a web-based UI for Kubernetes clusters. It is similar to [Kubernetes Dashboard](https://github.com/kubernetes/dashboard) in that it allow users to manage and troubleshoot applications running in the cluster and manage the cluster itself. However, Spectro Kubernetes Dashboard doesn’t require configuration. It’s easy to enable and it integrates seamlessly with Spectro Proxy.

Author: Revised this description.

| enableInsecureLogin | Enables non-Secure Sockets Layer (SSL) login. Dashboard login is always restricted to HTTP(S) + localhost or HTTPS and external domain. | `false` |
| ingress:enabled | Ingress configuration to access the ClusterIP, loadBalancer, or nodePort. | `false` |

## Usage

To use the Spectro Kubernetes Dashboard pack, you have to add it to your cluster profile. The Spectro Kubernetes Dashboard pack has default settings for **Access** and **Identity Provider**. These settings automatically install the latest version of the Spectro Proxy pack when the cluster is created.

Changing these default settings may require some additional configuration.
<br /> 

### Access

The default setting is **Private**. If you change the setting to **Public** and your cluster is in a public cloud, there is no additional configuration.  

However, if you change **Access** to **Public** and your cluster is in a private cloud, you have to manually add and configure the Spectro Proxy pack to your cluster profile. For more information, refer to the [Spectro Proxy](https://docs.spectrocloud.com/integrations/frp) guide.
<br /> 

### Identity Provider 

> The default setting is **Palette**. If you change this setting to **Inherit from Tenant**, you must configure OpenID Connect (OIDC) authentication. You can configure OIDC two ways:

Reviewer: Let's explain what using Palette as a value actually means. When you select Palette it means Palette becomes the identity provider, so any user that has a Palette account in the tenant and the proper permissions to view and access the respective project's resources will be able to log into the dashboard.

Author: Added.

> - In Tenant Admin scope, navigate to **Tenant Settings > SSO**, choose **OIDC**, and provide your third-party Identity Provider details.  

Reviewer: Let's link to the Palette SSO page. https://docs.spectrocloud.com/user-management/saml-sso

Author: Added link.

- In **Tenant Settings > SSO**, if you choose **None** or **SAML**, you must configure OIDC manually in the Kubernetes pack. 

### Enable OIDC Authentication

> To enable OIDC authentication manually, copy the configuration shown in the example to the Kubernetes Pack and provide your third-party Identity Provider details.

Reviewer: These parameters are already in the Kubernetes pack YAML definition. The user just needs to uncomment them, so there is no need to copy them.

Author: @karl-cardenas-coding I don't see the "OIDC related config" in the config file. 

@ravi-k8 @shantanubansal Can you please confirm? If a user changes the default value to 'Inherit from tenant', do they have to add this "OIDC related config" block?

Reviewer 2: @ritawatson I think the label should be "Inherit from Org"
If user selects "Inherit from Org" then backend would take care of adding client config block at run time. User is not required to add additional configuration.

Author: @ravi-k8 Has the UI label been changed? I'm going by the demo that was given. The UI at the time said 'Inherit from Tenant'.

Reviewer 2: @ritawatson Yes, label is updated to `Inherit from Organization`

<br />

> ```

Reviewer: This configuration is only applicable to EKS. Ideally, we should showcase the same for Azure and GCP. This is where Ravi suggested we ask the Q/A team. Arvind and Resma are two names that come to mind.

Author: @ravi-k8 @shantanubansal Do either of you have the OIDC configuration for Azure and GCP to include in the doc?

Reviewer 3: @ritawatson since we are providing a smooth and enhanced experience to user, as a part of that configuration is not directly visible to the user. But still In documentation we can give some configuration on How to Enable OIDC through our pack for different clouds.

@ritawatson @karl-cardenas-coding  maybe with the help of content team and testing team configuration also can be summarized in documentation for all the cloud types we support.

Author: @shantanubansal I need your help/information about how users can enable OIDC through the Spectro K8s Dashboard for different clouds. Can you please provide some details?

---
title: 'Spectro Kubernetes Dashboard'
metaTitle: "Spectro Cloud's Integrated Kubernetes Dashboard"
metaDescription: 'The Integrated Kubernetes Dashboard Monitoring pack in Palette reduces the complexity of standing up the Kubernetes dashboard for a cluster'
hiddenFromNav: true
type: "integration"
category: ['integrations','monitoring']
logoUrl: 'https://registry.spectrocloud.com/v1/k8s-dashboard/blobs/sha256:2de5d88b2573af42d4cc269dff75744c4174ce47cbbeed5445e51a2edd8b7429?type=image/png'
---

import Tabs from 'shared/components/ui/Tabs';
import WarningBox from 'shared/components/WarningBox';
import InfoBox from 'shared/components/InfoBox';
import PointsOfInterest from 'shared/components/common/PointOfInterest';
import Tooltip from "shared/components/ui/Tooltip";

# Spectro Kubernetes Dashboard

> Spectro Kubernetes Dashboard is a web-based UI for Kubernetes clusters that auto-enables the Kubernetes Dashboard. It integrates seamlessly with Spectro Proxy and uses default settings for access control and Identify Provider (IDP).   

Reviewer: ```suggestion
Spectro Kubernetes Dashboard is a web-based UI for Kubernetes clusters that auto-enables the Kubernetes Dashboard. It integrates seamlessly with [Spectro Proxy](/integrations/frp) and uses default settings for access control and Identity Provider (IDP).   
```
Typo, and added a link to the proxy to help the reader.

Author: Fixed typo and added link.

| Parameter | Description | Default |
|-----------|-------------|---------|
| Access | Controls cluster access. Private access means that nodes and pods are isolated from the internet. | Private |
| Identity Provider | You can use Palette as the IDP or a third-party application as the IDP. | Palette |

#### Internal Manifest Parameters

| Parameter | Description | Default |
|-----------|-------------|---------|
| namespace | The Kubernetes namespace to install the dashboard. | `kubernetes-dashboard` |
| ClusterRole | The ClusterRole to assign to the Kubernetes Dashboard. | `read-only` |
| certDuration | Self-signed certificate duration in hours. | 8760h (365d) |
| certRenewal | Self-signed certificate renewal in hours | 720h (30d) |
| enableInsecureLogin | RBAC ensures secure login. | `false` |
| serviceType | The service type for the dashboard. Supported values are ClusterIP, LoadBalancer, and NodePort. | `ClusterIP` |
| skipLogin | Enables or disables the skip login option on the dashboard. | `false` |
| enableInsecureLogin | Enables non-Secure Sockets Layer (SSL) login. Dashboard login is always restricted to HTTP(S) + localhost or HTTPS and external domain. | `false` |

> | ingress:enabled | Ingress configuration to access the ClusterIP, loadBalancer, or nodePort. | `false` |

Reviewer: ```suggestion
| ingress.enabled | Ingress configuration to access the ClusterIP, loadBalancer, or nodePort. | `false` |
```
Dot notation is more common to indicate a sub-section of a yaml or JSON object.

Author: Fixed.

## Usage

To use the Spectro Kubernetes Dashboard pack, you have to add it to your cluster profile. This integrated pack supports the following clouds:

- Amazon Web Services (AWS)
- Amazon Elastic Kubernetes Service (EKS)
- Azure
- Canonical Metal as a Service (MAAS) 
- Google Cloud Platform (GCP)
- OpenStack
- VMware vSphere

> 

Author: @ravi-k8 Is this list complete?

<br /> 

#### Identity Provider 

The default setting is **Palette**.

<br />

- **Palette**: No configuration is needed. This setting makes Palette the IDP, so any user with a Palette account in the tenant and the proper permissions to view and access the project's resources is able to log into the Kubernetes dashboard.

- **Inherit from Organization**: This setting requires you to configure OpenID Connect (OIDC) in Tenant Settings. In Tenant Admin scope, navigate to **Tenant Settings > SSO**, choose **OIDC**, and provide your third-party IDP details. For more information, check out the [SSO Setup](/user-management/saml-sso) guide. 

- **None**: This setting requires you to configure OIDC manually in the Kubernetes pack.

### Manually Configure OIDC

You only need to configure OIDC manually if you change the **Identify Provider** setting to **None**. The basic method to enable OIDC can be used for all cloud services except Amazon EKS. 

<br />

<Tabs>

<Tabs.TabPane tab="Basic OIDC Setup" key="Other Clouds">

To configure OIDC manually for clusters managed by most cloud providers, follow these steps: 

<br />

> 1. Copy ``oidc-`` configuration lines in the example and add them to the Kubernetes pack under the ``extraArgs`` parameter section. Enter your third-party provider details in quotes. <br /><br />

Reviewer: ```suggestion
1. Copy ``oidc-`` configuration lines in the following code snippet and add them to the Kubernetes pack under the ``extraArgs`` parameter section. Enter your third-party provider details in quotes. <br /><br />
```
I got confused when reading the example and thought I missed something 😵‍💫 . This could just be me, but I think specifying the code snippet will reduce the chances this happening to someone else.

  ```
  kubeadmconfig:
    apiServer:
      extraArgs:
      oidc-issuer-url: "provider URL"
      oidc-client-id: "client-id"
      oidc-groups-claim: "groups"
      oidc-username-claim: "email"
  ```

2. Under the ``clientConfig`` parameter section of Kubernetes pack, uncomment the ``oidc-`` configuration lines, and enter your provider details in quotes. The provider URL and client-id must be the same in the ``extraArgs`` and ``clientConfig`` parameter sections. <br /><br />

  ```
    clientConfig:
      oidc-issuer-url: "provider URL"
      oidc-client-id: "client-id"
      oidc-client-secret: client-secret-value
      oidc-extra-scope: profile,email,openid
    ```

<br />

</Tabs.TabPane>

<Tabs.TabPane tab="AWS EKS" key="AWS EKS">

To enable OIDC manually for EKS clusters, follow these steps: 

<br />

1. In the Kubernetes pack, uncomment the lines in the ``oidcIdentityProvider`` parameter section of the Kubernetes pack, and enter your third-party provider details.
<br />

  ```

>     oidcIdentityProvider:
>       identityProviderConfigName: 'Spectro-docs'     # The name of the OIDC provider configuration
>       issuerUrl: 'issuer-url'       # The URL of the OpenID identity provider
>       clientId: 'user-client-id-from-Palette'           # The ID for the client application that makes authentication requests to the OpenID identity provider
>       usernameClaim: "email"                     # The JSON Web Token (JWT) claim to use as the username
>       usernamePrefix: "-"                        # The prefix that is prepended to username claims to prevent clashes with existing names
>       groupsClaim: "groups"                      # The JWT claim that the provider uses to return your groups
>       groupsPrefix: ""                          # The prefix that is prepended to group claims to prevent clashes with existing names
>       requiredClaims:                            # The key value pairs that describe required claims in the identity token

Reviewer: ```suggestion
    oidcIdentityProvider:
      identityProviderConfigName: "Spectro-docs" 
      issuerUrl: "issuer-url"    
      usernameClaim: "email"     
      usernamePrefix: "-"      
      groupsClaim: "groups"      
      groupsPrefix: ""                          
      requiredClaims:  ""                          
```
I removed the comments to improve the readability and make it easier for people to copy-paste the code snippet.

However, if you change **Access** to **Public** and your cluster is in a private cloud, you have to manually add and configure the Spectro Proxy pack to your cluster profile. For more information, refer to the [Spectro Proxy](/integrations/frp) guide.

<br /> 

### Identity Provider 

The default setting is **Palette**.

<br />

- **Palette**: No configuration is required. This setting makes Palette the IDP, so any user with a Palette account in the tenant and the proper permissions to view and access the project's resources is able to log into the Kubernetes dashboard.

- **Inherit from Organization**: This setting requires you to configure OIDC authentication in Tenant Settings. In Tenant Admin scope, navigate to **Tenant Settings > SSO**, choose **OIDC**, and provide your third-party IDP details. For more information, check out the [SSO Setup](/user-management/saml-sso) guide. 

- **None**: For all cloud services except Azure AKS, this setting requires you to configure OpenID Connect (OIDC) authentication. Azure AKS clusters require authentication through Active Directory (AD).

> 

Author: @MGamya @arvind-sys 
Are descriptions for **Inherit from Organization** and **None** accurate?

Reviewer 2: @ritawatson We don't need to specify the Azure AKS, as we will have different section to list supported clouds to have spectro k8s dashboard integration.

Author: @ravi-k8 Removed Azure AKS throughout.

      usernameClaim: "email"                     # The JSON Web Token (JWT) claim to use as the username
      usernamePrefix: "-"                        # The prefix that is prepended to username claims to prevent clashes with existing names
      groupsClaim: "groups"                      # The JWT claim that the provider uses to return your groups
      groupsPrefix: ""                          # The prefix that is prepended to group claims to prevent clashes with existing names
      requiredClaims:                            # The key value pairs that describe required claims in the identity token
  ```

<br />

2. Under the ``clientConfig`` parameter section of Kubernetes pack, uncomment the ``oidc-`` configuration lines. <br /><br />

  ```
  clientConfig:
    oidc-issuer-url: "{{ .spectro.pack.kubernetes-eks.managedControlPlane.oidcIdentityProvider.issuerUrl }}"
    oidc-client-id: "{{ .spectro.pack.kubernetes-eks.managedControlPlane.oidcIdentityProvider.clientId }}"
    oidc-client-secret: 1gsranjjmdgahm10j8r6m47ejokm9kafvcbhi3d48jlc3rfpprhv
    oidc-extra-scope: profile,email
  ``` 

</Tabs.TabPane>

<Tabs.TabPane tab="Azure AKS" key="Azure AKS">

>  

Author: @MGamya @arvind-sys 
Please review this AKS section carefully. Are all the necessary steps included? What is the command in step 5 all about? What does it do?

Author: Question no longer applies. Azure AKS tab is removed.

<br />

3. In Palette, create a cluster profile using the Spectro RBAC add-on pack available under Authentication packs in the Public Repo. Check out [Creating Cluster Profiles](/cluster-profiles/task-define-profile) to learn about cluster profiles.
<br />

4. Add information in the Spectro RBAC pack for users and groups and their Roles and RoleBindings based on the same information in Azure AD: <br /><br /> 

  ```
  clusterRoleBindings:
      - role: admin
        name: special-override-name
        subjects:
        - {type: User, name: user5}
        - {type: Group, name: group5}
      - role: view
        subjects:
        - {type: User, name: user6}
        - {type: Group, name: group6}
        - {type: ServiceAccount, name: group6, namespace: foo}
      namespaces:
      - namespace: team1
        createNamespace: true
        roleBindings:
      - role: admin
        name: special-override-name-role
        kind: ClusterRole
        subjects:
        - {type: User, name: user3}
        - {type: Group, name: group3}
      - role: view
        kind: ClusterRole
        subjects:
        - {type: User, name: user4}
        - {type: Group, name: group4}

>   ```

Author: @MGamya @arvind-sys 
Is the above example correct?

Author: Question no longer applies. Azure AKS tab is removed.

<br /> 

5. To get the user-specific kubeconfig file, run the following command:

>   ``az aks get-credentials --resource-group <resource-group> --name <cluster-name>``

Author: @MGamya @arvind-sys
The above command is in the original documentation. 
- Why would users run the command? 
- What does the command do?

Author: Question no longer applies - Azure AKS tab is removed.
